Stratis.Bitcoin.Features.LightWallet/LightWalletFeePolicy.cs

```diff
@@ -29,29 +30,63 @@ public LightWalletFeePolicy(IAsyncLoopFactory asyncLoopFactory, INodeLifetime no
             this.highTxFeePerKb = null;
             this.mediumTxFeePerKb = null;
             this.lowTxFeePerKb = null;
-            this.httpClient = new HttpClient();
             this.asyncLoopFactory = asyncLoopFactory;
             this.nodeLifetime = nodeLifetime;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.maxTxFee = new Money(0.1M, MoneyUnit.BTC);
+            this.initializedOnce = false;
         }
 
         public Task Initialize()
         {
             return this.asyncLoopFactory.Run(nameof(LightWalletFeePolicy), async token =>
             {
-                HttpResponseMessage response =
-                    await this.httpClient.GetAsync(@"http://api.blockcypher.com/v1/btc/main", HttpCompletionOption.ResponseContentRead, token)
-                    .ConfigureAwait(false);
+                // This will run evry 3 to 10 minutes randomly
+                // So the API provider is not able to identify our transaction with a timing attack
+                var waitMinutes = new Random().Next(3, 10);
+
+                HttpResponseMessage response = null;
+                try
+                {
+                    // TestNet fee estimation is useless, because the miners don't select transactions, like mainnet miners
+                    // Test results on RegTest and TestNet are more illustrative with mainnet fees
+                    response =
+                        await HttpClient.GetAsync(@"http://api.blockcypher.com/v1/btc/main", HttpCompletionOption.ResponseContentRead, token)
+                        .ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                    return;
+                }
+                catch (Exception ex) when (ex is HttpRequestException || ex is TimeoutException)
+                {
+                    response = null;
+                }
+                if (response == null || !response.IsSuccessStatusCode)
+                {
+                    // If it's already been initialized once just keep using the feerate we already have
+                    if (this.initializedOnce)
+                    {
+                        await Task.Delay(TimeSpan.FromMinutes(waitMinutes), token).ContinueWith(t => { }).ConfigureAwait(false);
```

Comment:
can't this be just 
`await Task.Delay(TimeSpan.FromMinutes(waitMinutes), token).ConfigureAwait(false);`

Reply:
To swallow exception